### ISSUE TITLE
match with tuples is not properly detected

### ISSUE THREAD

**Issue by Terseus** [astral-sh/ty#2708]
### Summary

Consider this code:

```python
from typing import assert_never


def func(value: int | tuple[int, int]) -> int:
    match value:
        case int(value):
            return value
        case (left, right):
            return left + right
        case _:
            assert_never(value)


def main():
    print(func(10))
    print(func((10, 20)))


if __name__ == "__main__":
    main()
```

If we execute it we can see it works:

```
❯ python example.py
10
30
```

However for `ty` is not valid:

```
❯ ty check example.py
error[type-assertion-failure]: Argument does not have asserted type `Never`
  --> example.py:11:13
   |
 9 |             return left + right
10 |         case _:
11 |             assert_never(value)
   |             ^^^^^^^^^^^^^-----^
   |                          |
   |                          Inferred type of argument is `tuple[int, int]`
   |
info: `Never` and `tuple[int, int]` are not equivalent types
info: rule `type-assertion-failure` is enabled by default

Found 1 diagnostic
```

It doesn't matter if we remove the parentheses around the values of the tuple match case.

To make `ty` happy we need to replace this:
```python
case (left, right):
```

with this:
```python
case tuple(left, right):
```

But now it doesn't work because when using a built-in type, Python expects to capture the whole value:
```
❯ python example.py
10
Traceback (most recent call last):
  File "/home/terseus/programacion/python/pyinfra-ufw/example.py", line 20, in <module>
    main()
    ~~~~^^
  File "/home/terseus/programacion/python/pyinfra-ufw/example.py", line 16, in main
    print(func((10, 20)))
          ~~~~^^^^^^^^^^
  File "/home/terseus/programacion/python/pyinfra-ufw/example.py", line 8, in func
    case tuple(left, right):
         ~~~~~^^^^^^^^^^^^^
TypeError: tuple() accepts 1 positional sub-pattern (2 given)
```

This is documented in the [PEP 622](https://peps.python.org/pep-0622/#the-match-protocol) (emphasis mine):
> For the most commonly-matched built-in types (bool, bytearray, bytes, dict, float, frozenset, int, list, set, str, and tuple), a single positional sub-pattern is allowed to be passed to the call. **Rather than being matched against any particular attribute on the subject, it is instead matched against the subject itself**.
---

I'm not sure if this is related to #561, it doesn't look like it because that issue handles type narrowing on tuple matches, while this problem happens even when not trying to narrow its types, only capture its members.

### Version

ty 0.0.14 (16597f5e2 2026-01-26)

**Comment by carljm** [comment]
Thanks for the report! You're right that this case is a bit different from the one currently described in #561 -- but I think it will still be most useful to combine them, because they will be addressed together. These are both different forms of narrowing the match subject type based on a tuple pattern, and they will almost certainly be implemented together.

Closing as duplicate of #561, but will also add your example to that issue.
